stanford_gallery.module

```diff
@@ -55,11 +55,18 @@ function stanford_gallery_field_formatter_view($entity_type, $entity, $field, $i
 
   foreach ($items as $delta => $item) {
     if ($field_collection = field_collection_field_get_entity($item)) {
+
+      // Patch up empty alt text.
+      if (empty($field_collection->field_s_image_image[LANGUAGE_NONE][0]["alt"]) && isset($field_collection->field_s_image_caption[LANGUAGE_NONE][0]["safe_value"])) {
```

Comment:
Won't the part `isset($field_collection->field_s_image_caption[LANGUAGE_NONE][0]["safe_value"])` throw an error if `$field_collection->field_s_image_caption` is an empty array or null?

Reply:
Shouldn't, that is what isset() protects against. 
http://php.net/manual/en/function.isset.php

I chose to use isset() instead of empty() in case there was a time where the caption was removed or unavailable. I am not checking if it is empty because if the alt text is empty and there is _something_ in the caption field I want to use that.

Reply:
https://stackoverflow.com/questions/1219542/in-where-shall-i-use-isset-and-empty

Reply:
hmm. for some reason i thought it would throw a notice or something. guess not. works for me